### ISSUE TITLE
`Style/GlobalStdStream` cop is unsafe

### ISSUE THREAD

**Issue by tsugimoto** [rubocop-hq/rubocop#8688]
because `$stdout` and `STDOUT` may point different object.

```ruby
$stdout = STDERR
STDOUT.puts($stdout == STDOUT) # false
```

## Expected behavior

Auto-corrects with `--auto-correct-all`, but not `--auto-correct`.

## Actual behavior

Auto-corrects with `--auto-correct`.

Example source code now prints `true` instead of `false`.

## Steps to reproduce the problem

`rubocop --auto-correct --only Style/GlobalStdStream`

## RuboCop version

```
$ bundle exec rubocop -V
0.90.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.6.3 x86_64-darwin18)
```

**Comment by marcandre** [comment]
Absolutely! Thanks :-)
Fixed in #8691.
